Regression for groovy routes

When I try to run the example I get:

```
[1] Monitoring pod routes-cfb6655dc-zvf7r[1] Starting the Java application using /opt/run-java/run-java.sh ...
[1] exec java -javaagent:/opt/prometheus/jmx_prometheus_javaagent.jar=9779:/opt/prometheus/prometheus-config.yml -XX:+UseParallelGC -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -XX:MinHeapFreeRatio=20 -XX:MaxHeapFreeRatio=40 -XX:+ExitOnOutOfMemoryError -cp .:/deployments/dependencies/org.apache.camel.k/camel-k-runtime-jvm-0.0.3-SNAPSHOT.jar:/deployments/dependencies/org.slf4j/slf4j-api-1.7.25.jar:/deployments/dependencies/org.apache.logging.log4j/log4j-slf4j-impl-2.11.0.jar:/deployments/dependencies/org.apache.logging.log4j/log4j-api-2.11.0.jar:/deployments/dependencies/org.apache.logging.log4j/log4j-core-2.11.0.jar:/deployments/dependencies/org.jooq/joor-java-8-0.9.9.jar:/deployments/dependencies/commons-io/commons-io-2.6.jar:/deployments/dependencies/org.apache.commons/commons-lang3-3.8.jar:/deployments/dependencies/org.apache.camel/camel-core-2.22.1.jar:/deployments/dependencies/com.sun.xml.bind/jaxb-core-2.3.0.1.jar:/deployments/dependencies/com.sun.xml.bind/jaxb-impl-2.3.0.1.jar org.apache.camel.k.jvm.Application
[1] [INFO ] 2018-09-24 16:58:57.113 [main] Application - Routes: file:/etc/camel/conf/routes.groovy
[1] [INFO ] 2018-09-24 16:58:57.115 [main] Application - Language: groovy
[1] [INFO ] 2018-09-24 16:58:57.115 [main] Application - Locations: file:/etc/camel/conf/application.properties
[1] Exception in thread "main" java.lang.NoClassDefFoundError: org/codehaus/groovy/control/CompilerConfiguration
[1]     at org.apache.camel.k.jvm.RoutesLoaders$4$1.configure(RoutesLoaders.java:138)
[1]     at org.apache.camel.builder.RouteBuilder.checkInitialized(RouteBuilder.java:462)
[1]     at org.apache.camel.builder.RouteBuilder.configureRoutes(RouteBuilder.java:402)
[1]     at org.apache.camel.builder.RouteBuilder.addRoutesToCamelContext(RouteBuilder.java:383)
[1]     at org.apache.camel.impl.DefaultCamelContext$1.call(DefaultCamelContext.java:1029)
[1]     at org.apache.camel.impl.DefaultCamelContext$1.call(DefaultCamelContext.java:1026)
[1]     at org.apache.camel.impl.DefaultCamelContext.doWithDefinedClassLoader(DefaultCamelContext.java:3272)
[1]     at org.apache.camel.impl.DefaultCamelContext.addRoutes(DefaultCamelContext.java:1026)
[1]     at org.apache.camel.main.MainSupport.postProcessCamelContext(MainSupport.java:612)
[1]     at org.apache.camel.main.MainSupport.postProcessContext(MainSupport.java:550)
[1]     at org.apache.camel.main.Main.doStart(Main.java:136)
[1]     at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61)
[1]     at org.apache.camel.main.MainSupport.run(MainSupport.java:170)
[1]     at org.apache.camel.k.jvm.Application.main(Application.java:68)
[1] Caused by: java.lang.ClassNotFoundException: org.codehaus.groovy.control.CompilerConfiguration
[1]     at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
[1]     at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
[1]     at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
[1]     at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
[1]     ... 14 more
```


@oscerd , @zregvart do you think we have any chance to run a dev instance of Openshift in Apache CI?

Currently PR builds are disabled because there was a issue with dependencies, but I think it's easy to overcome that problem by checking-in the vendor dir... What about `minishift` / `oc cluster up` to avoid breaking things?

--- a/Gopkg.lock
+++ b/Gopkg.lock
@@ -493,14 +493,6 @@
   version = "v0.9.1"
 
 [[projects]]
-  branch = "v1"
-  digest = "1:52ffb9db0286de37253a5098607cfbcfcdc94e51e8c226da120513df82adab0c"
-  name = "gopkg.in/yaml.v1"
-  packages = ["."]
-  pruneopts = "NUT"
-  revision = "9f9df34309c04878acc86042b16630b0f696e1de"
-
-[[projects]]
   digest = "1:7c95b35057a0ff2e19f707173cc1a947fa43a6eb5c4d300d196ece0334046082"
   name = "gopkg.in/yaml.v2"
   packages = ["."]
@@ -741,7 +733,6 @@
     "github.com/spf13/cobra",
     "github.com/stoewer/go-strcase",
     "github.com/stretchr/testify/assert",
-    "gopkg.in/yaml.v1",
     "gopkg.in/yaml.v2",
     "k8s.io/api/apps/v1",
     "k8s.io/api/core/v1",

--- a/pkg/client/cmd/completion_bash.go
+++ b/pkg/client/cmd/completion_bash.go
@@ -66,6 +66,11 @@ __kamel_dependency_type() {
     esac
 }
 
+__kamel_languages() {
+    local type_list="js groovy java xml"
+    COMPREPLY=( $( compgen -W "${type_list}" -- "$cur") )
+}
+
 __kamel_kubectl_get_configmap() {
     local template
     local kubectl_out
@@ -181,6 +186,13 @@ func configureKnownBashCompletions(command *cobra.Command) {
 		"context",
 		map[string][]string{
 			cobra.BashCompCustom: {"__kamel_kubectl_get_user_integrationcontexts"},
+		},
+	)
+	configureBashAnnotationForFlag(
+		command,
+		"language",
+		map[string][]string{
+			cobra.BashCompCustom: {"__kamel_languages"},
 		},
 	)
 }

--- a/pkg/client/cmd/run.go
+++ b/pkg/client/cmd/run.go
@@ -25,6 +25,8 @@ import (
 	"strconv"
 	"strings"
 
+	"github.com/apache/camel-k/pkg/util"
+
 	"github.com/apache/camel-k/pkg/util/sync"
 	"github.com/sirupsen/logrus"
 
@@ -265,6 +267,15 @@ func (o *runCmdOptions) updateIntegrationCode(filename string) (*v1alpha1.Integr
 		} else if strings.HasPrefix(item, "camel-") {
 			integration.Spec.Dependencies = append(integration.Spec.Dependencies, "camel:"+strings.TrimPrefix(item, "camel-"))
 		}
+	}
+
+	// special handling for groovy
+	// TODO: we should define handlers for languages and/or file extensions
+	if o.Language == "groovy" && !util.StringSliceExists(o.Dependencies, "camel:groovy") {
+		integration.Spec.Dependencies = append(integration.Spec.Dependencies, "camel:groovy")
+	}
+	if o.Language == "" && strings.HasSuffix(filename, ".groovy") {
+		integration.Spec.Dependencies = append(integration.Spec.Dependencies, "camel:groovy")
 	}
 
 	for _, item := range o.Properties {

--- a/pkg/stub/action/integration/build.go
+++ b/pkg/stub/action/integration/build.go
@@ -19,6 +19,8 @@ package action
 
 import (
 	"fmt"
+
+	"github.com/apache/camel-k/pkg/util"
 	"github.com/apache/camel-k/pkg/util/digest"
 
 	"github.com/rs/xid"
@@ -60,7 +62,7 @@ func (action *buildAction) Handle(integration *v1alpha1.Integration) error {
 			// amended to add/remove dependencies
 
 			//TODO: this is a very simple check, we may need to provide a deps comparison strategy
-			if !StringSliceContains(ctx.Spec.Dependencies, integration.Spec.Dependencies) {
+			if !util.StringSliceContains(ctx.Spec.Dependencies, integration.Spec.Dependencies) {
 				// We need to re-generate a context or search for a new one that
 				// satisfies integrations needs so let's remove the association
 				// with a context

--- a/pkg/stub/action/integration/util.go
+++ b/pkg/stub/action/integration/util.go
@@ -4,6 +4,8 @@ import (
 	"fmt"
 	"strings"
 
+	"github.com/apache/camel-k/pkg/util"
+
 	"github.com/apache/camel-k/pkg/apis/camel/v1alpha1"
 	"github.com/operator-framework/operator-sdk/pkg/sdk"
 	"k8s.io/api/core/v1"
@@ -38,35 +40,13 @@ func LookupContextForIntegration(integration *v1alpha1.Integration) (*v1alpha1.I
 				continue
 			}
 
-			if StringSliceContains(ctx.Spec.Dependencies, integration.Spec.Dependencies) {
+			if util.StringSliceContains(ctx.Spec.Dependencies, integration.Spec.Dependencies) {
 				return &ctx, nil
 			}
 		}
 	}
 
 	return nil, nil
-}
-
-// StringSliceContains --
-func StringSliceContains(slice []string, items []string) bool {
-	for i := 0; i < len(items); i++ {
-		if !StringSliceExists(slice, items[i]) {
-			return false
-		}
-	}
-
-	return true
-}
-
-// StringSliceExists --
-func StringSliceExists(slice []string, item string) bool {
-	for i := 0; i < len(slice); i++ {
-		if slice[i] == item {
-			return true
-		}
-	}
-
-	return false
 }
 
 // PropertiesString --

--- a/pkg/util/util.go
+++ b/pkg/util/util.go
@@ -1,0 +1,40 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+// StringSliceContains --
+func StringSliceContains(slice []string, items []string) bool {
+	for i := 0; i < len(items); i++ {
+		if !StringSliceExists(slice, items[i]) {
+			return false
+		}
+	}
+
+	return true
+}
+
+// StringSliceExists --
+func StringSliceExists(slice []string, item string) bool {
+	for i := 0; i < len(slice); i++ {
+		if slice[i] == item {
+			return true
+		}
+	}
+
+	return false
+}

